### ISSUE TITLE
Source map not uploaded to sentry and debug:true crashing build

### ISSUE THREAD

**Issue by kopax-polyconseil** [getsentry/sentry-webpack-plugin#332]
- [x] Review the documentation: https://docs.sentry.io/
- [x] Search for existing issues: https://github.com/getsentry/sentry-javascript/issues
- [x] Use the latest release: https://github.com/getsentry/sentry-javascript/releases
- [x] Provide a link to the affected event from your Sentry account

## Package + Version

- [x] `@sentry/react`: `6.13.3`
- [x] `@sentry/webpack-plugin`: `1.18.3`
- [x] `@sentry/tracing`: `6.13.3`

### Reproduction:

Repository https://github.com/pass-culture/pass-culture-app-native

```
 yarn build:testing
yarn run v1.22.15
$ ENV=testing yarn build
$ node web/scripts/build.js
Creating an optimized production build...
(node:25556) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.5.2

Please only submit bug reports when using the officially supported version.

=============
Compiled successfully.

File sizes after gzip:

  1 MB       build/static/js/2.acaa97ce.chunk.js
  458.42 KB  build/static/js/main.221745e8.chunk.js
  1.73 KB    build/static/js/4.4b36eeca.chunk.js
  1.34 KB    build/static/js/3.bdadd5cd.chunk.js
  1.2 KB     build/static/js/runtime-main.1d547549.js

The bundle size is significantly larger than recommended.
Consider reducing it with code splitting: https://goo.gl/9VhYWB
You can also analyze the project dependencies: https://goo.gl/LeUzfb

The project was built assuming it is hosted at https://app.testing.passculture.team/.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  https://cra.link/deployment

Done in 184.04s
```

Source maps are well generated by webpack:

```bash
$ find build/static -type f -name '*.map'
build/static/js/4.4b36eeca.chunk.js.map
build/static/js/3.bdadd5cd.chunk.js.map
build/static/js/2.acaa97ce.chunk.js.map
build/static/js/runtime-main.1d547549.js.map
build/static/js/main.221745e8.chunk.js.map
```

Looking at our sentry, we do not have them: 

![image](https://user-images.githubusercontent.com/77674046/143889398-66009b87-3950-4574-a5cc-4310914044ca.png)

This is the whole [`webpack.config.js`](https://github.com/pass-culture/pass-culture-app-native/blob/master/web/config/webpack.config.js#L130)

## Description

`@sentry/webpack-plugin` does not upload source maps, nor add logging anywhere.

I was able to use `debug: true` option but I add to edit `sentry-webpack-plugin` main `index.js` file:

```diff
    const input = {
      loaders: sillyClone(
-        compilerOptions.module.loaders || compilerOptions.module.rules
+        compilerOptions.module.loaders || compilerOptions.module.rules || []
      ).map(getLoaderName),
      entry: sillyClone(compilerOptions.entry),
    };

    this.injectRelease(compilerOptions);

    const output = {
      loaders: sillyClone(
-        compilerOptions.module.loaders || compilerOptions.module.rules
+        compilerOptions.module.loaders || compilerOptions.module.rules || []
      ).map(getLoaderName),
      entry: sillyClone(compilerOptions.entry),
    };
```

This is the debug output:

```
$ yarn build:testing
yarn run v1.22.15
$ ENV=testing yarn build
$ node web/scripts/build.js
Creating an optimized production build...
[Sentry Webpack Plugin] DEBUG: Injecting release code
[Sentry Webpack Plugin] DEBUG: Loaders:
 [
  '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry.loader.js'
]
[Sentry Webpack Plugin] DEBUG: Added loaders:
 [
  '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry.loader.js'
]
[Sentry Webpack Plugin] DEBUG: Removed loaders:
 []
[Sentry Webpack Plugin] DEBUG: Entry:
 '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js'
[Sentry Webpack Plugin] DEBUG: Added entry:
 [
  '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js'
]
[Sentry Webpack Plugin] DEBUG: Removed entry:
 [ undefined ]
(node:9576) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.5.2

Please only submit bug reports when using the officially supported version.

=============
[Sentry Webpack Plugin] DEBUG: Injecting release code
[Sentry Webpack Plugin] DEBUG: Loaders:
 [
  '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry.loader.js'
]
[Sentry Webpack Plugin] DEBUG: Added loaders:
 [
  '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry.loader.js'
]
[Sentry Webpack Plugin] DEBUG: Removed loaders:
 []
[Sentry Webpack Plugin] DEBUG: Entry:
 '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js'
[Sentry Webpack Plugin] DEBUG: Added entry:
 [
  '/home/dka/workspace/github.com/pass-culture/pass-culture-app-native/node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js'
]
[Sentry Webpack Plugin] DEBUG: Removed entry:
 [ undefined ]
Compiled successfully.

File sizes after gzip:

  1 MB       build/static/js/2.acaa97ce.chunk.js
  458.42 KB  build/static/js/main.5d91f0a3.chunk.js
  1.73 KB    build/static/js/4.4b36eeca.chunk.js
  1.34 KB    build/static/js/3.bdadd5cd.chunk.js
  1.2 KB     build/static/js/runtime-main.1d547549.js

The bundle size is significantly larger than recommended.
Consider reducing it with code splitting: https://goo.gl/9VhYWB
You can also analyze the project dependencies: https://goo.gl/LeUzfb

The project was built assuming it is hosted at https://app.testing.passculture.team/.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  https://cra.link/deployment

Done in 132.32s.

```

**Comment by kopax-polyconseil** [comment]
Hello, I believe this can help find the root cause for why am I not able to upload source maps, so please allow me to ask if there is any update for this?

**Comment by AbhiPrasad** [comment]
Hey, it seems you are using `@sentry/react-native` for both react native web and react native mobile (as per: https://github.com/pass-culture/pass-culture-app-native/blob/a10e0efaa49358def2dc04b3d7e48c9148c992b0/src/libs/monitoring/services.ts#L1).

You might have to [manually upload your source maps with `sentry-cli`](https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/cli/). Will backlog this issue so that we can investigate react native web further though.

See the React Native source maps instructions here: https://docs.sentry.io/platforms/react-native/sourcemaps.

**Comment by kopax-polyconseil** [comment]
@AbhiPrasad we are not, we are using webpack `resolve.aliases` here https://github.com/pass-culture/pass-culture-app-native/blob/a10e0efaa49358def2dc04b3d7e48c9148c992b0/web/config/webpack.config.js#L299

Perhaps this plugin does not know about aliases ?

**Comment by kopax-polyconseil** [comment]
Hi and happy new year @AbhiPrasad , do you got any update on this one ?

**Comment by kamilogorek** [comment]
I wonder how it's possible that `compilerOptions.module` doesn't have any valid `loaders` or `rules`, especially that they are present in your webpack config. It may something to do with PnP and `PnpWebpackPlugin.moduleLoader(module)` block.

Can you instead of adding `[]` as fallbacks, debug what's actually present under `compilerOptions`?

As for the initial issues, there should be a regular upload summary, which looks like is not printed out? If so, it means that there are no files found, which can indicate that webpack plugin is run before assets are compiled?

**Comment by kopax-polyconseil** [comment]
Thanks for your reply, I was wondering how we could solve this, I will dig into it in the next weeks and I will let you know the results.

**Comment by kopax-polyconseil** [comment]
Hi @kamilogorek , I had a  try to enable `debug: true` and look for `compilerOptions`.

I found their declaration here : https://github.com/pass-culture/pass-culture-app-native/blob/master/tsconfig.json#L2

I added :  

- `console.log('1', JSON.stringify(compilerOptions, null, 2))` before the return here : https://github.com/pass-culture/pass-culture-app-native/blob/master/web/config/pnpTs.js#L5
- `console.log('2', JSON.stringify(compilerOptions, null, 2))` before the return here : https://github.com/pass-culture/pass-culture-app-native/blob/master/web/config/pnpTs.js#L9

Running the build break exactly the same as before. I also tried to move the `SentryWebpackPlugin` after the `PnpWebpackPlugin` and after the `ModuleScopePlugin` so basically here : https://github.com/pass-culture/pass-culture-app-native/blob/master/web/config/webpack.config.js#L354 , and it didn't help.

Can you clarify how I should debug the `compilerOptions` ?

**Comment by kamilogorek** [comment]
@kopax-polyconseil spent some time to resolve this. Your webpack config is just incorrect, so no need to fix anything on the CLI side.

Move `SentryWebpackPlugin` from `release.plugins` field, to just `plugins`, which you list at the bottom. `release.plugins` is going through different set of compilation hooks, so we cannot inject into assets emitting process.
Once you move it, it will work just fine, confirmed locally.

Cheers!

**Comment by kopax-polyconseil** [comment]
Thanks @kamilogorek this really helped. However I am still stuck having the source map being useful after being uploaded to sentry. I have tried a bunch of settings including `"sourceRoot": "/",` in `tsconfig.ts`, and a bunch of settings for `include.ext`, `include.paths`, `ignore`, `sourceMapReference`, `cleanArtifacts`, `stripCommonPrefix`, I do not get if I:

- should add `tsx` and `ts` to `ext`
- should remove `js` from `ext`
- should ignore `*.native.*`, `*.ios.*`, `*.android.*`, `*/__tests__/*`, `*/__mocks__/*`, `*.test.*`, I have tried and it did not work using `.gitignore` ignore syntaxe.
- should  set the `urlPrefix`, `urlSuffix` or `stripPrefix` 
- consider the path in the `.map` file from `src` and `tsx` , so far it just point like if root was `src`, while build is in `build`

We hope that we can enjoy sentry with our web application and thanks a lot for helping us achieve a working configuration, it is well appreciated !
